experimental/Schemes/elliptic_surface.jl

```diff
@@ -0,0 +1,1116 @@
+    local redfib = collect(keys(factor(disc).fac))
+    push!(refined_charts, PrincipalOpenSubset(W, redfib))
+  end
+
+  # no extra singularities on the the zero section
+  # This is the Y = 1 chart
+  # therefore we just exclude all the singularities visible here
+  for W in [P[1][3],P[1][6]]
+    local Ising = I_sing_P(W)
+    if isone(Ising)
+      push!(refined_charts, W)
+      continue
+    end
+    local (z,x,s_or_t) = coordinates(W)
+    # reducible singular fibers
+    local disc = gens(eliminate(Ising, [x, s_or_t]))[1]
+    local redfib = collect(keys(factor(disc).fac))
+    push!(refined_charts, PrincipalOpenSubset(W, redfib))
+  end
+
+
+  Cref = Covering(refined_charts)
+  inherit_glueings!(Cref, P[1])
+  push!(P.coverings, Cref)
+  @assert has_decomposition_info(default_covering(P))
+  inherit_decomposition_info!(P, Cref)
+  @assert has_decomposition_info(Cref)
+  # Now we have an extra covering where each chart just contains a single singularity
+
+  @assert scheme(I_sing) === S
+  @assert scheme(I_sing_P) === P
+  return Cref
+end
+
+@doc raw"""
+    relatively_minimal_model(E::EllipticSurface) -> CoveredScheme, SchemeMor
+
+Return the relatively minimal model $X \to C$ and the contraction
+$\Psi \colon X \to S$ to its weierstrass model $S$.
+"""
+function relatively_minimal_model(E::EllipticSurface)
+  if isdefined(E, :blowups)
+    return E.Y, E.blowup
+  end
+  S, inc_S = weierstrass_model(E)
+  Crefined = _separate_singularities!(E)
+  # Blow up singular points (one at a time) until smooth
+  # and compute the strict transforms of the `divisors`
+  # collect the exceptional divisors
+  # blowup ambient spaces: X0 → X   ⊂
+  # blowup pi: (K3 = Y0)  → (S singular weierstrass model)
+  #
+  # initialization for the while loop
+  X0 = codomain(inc_S)
+  Y0 = S
+  inc_Y0 = inc_S
+
+  exceptionals = []
+  varnames = [:a,:b,:c,:d,:e,:f,:g,:h,:i,:j,:k,:l,:m,:n,:o,:p,:q,:r,:u,:v,:w]
+  projectionsX = []
+  projectionsY = []
+  count = 0
+
+  @vprint :EllipticSurface 2 "Blowing up Weierstrass model\n"
+  @vprint :EllipticSurface 2 "in $(Crefined)\n"
+  while true
+    count = count+1
+    @vprint :EllipticSurface 1 "blowup number: $(count)\n"
+    @vprint :EllipticSurface 2 "computing singular locus\n"
+    I_sing_Y0 = ideal_sheaf_of_singular_locus(Y0)
+    @vprint :EllipticSurface 2 "decomposing singular locus\n"
+    I_sing_Y0 = maximal_associated_points(I_sing_Y0)
+    I_sing_X0 = pushforward(inc_Y0).(I_sing_Y0)
+    @vprint :EllipticSurface 1 "number of singular points: $(length(I_sing_Y0))\n"
+    if length(I_sing_Y0)==0
+      # stop if smooth
+      break
+    end
+    # make sure there is only one singular point per chart
+    if count == 1
+      cov = Crefined
+    else
+      # the following leads to difficult bugs
+      #cov0 = simplified_covering(X0)
+      #cov1 = _separate_disjoint_components(I_sing_X0, covering=cov0)
+      #cov = _one_patch_per_component(cov1, I_sing_X0)
+      #push!(X0.coverings, cov)
```

Comment:
Does this still hold? Then we should look into this.

Reply:
```
ERROR: patch not found
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] _flatten_open_subscheme(U::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}, C::Covering{QQField}; iso::SpecMor{Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}, PrincipalOpenSubset{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}, Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}}, MPolyQuoLocalizedRingHom{MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}, Oscar.MPolyAnyMap{QQMPolyRing, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}, Nothing, MPolyQuoLocRingElem{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}}})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/SimplifiedSpec.jl:154
  [3] _flatten_open_subscheme(U::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}, C::Covering{QQField})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/SimplifiedSpec.jl:150
  [4] (::Oscar.var"#restriction_func#5232"{CoveredScheme{QQField}})(F::PreSheafOnScheme{CoveredScheme{QQField}, Union{AbsSpec, SpecOpen}, Ring, Map}, V::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}, U::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/Types.jl:301
  [5] restriction_map(F::PreSheafOnScheme{CoveredScheme{QQField}, Union{AbsSpec, SpecOpen}, Ring, Map}, U::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}, V::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}; check::Bool)
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/Sheaves.jl:111
  [6] restriction_map(F::PreSheafOnScheme{CoveredScheme{QQField}, Union{AbsSpec, SpecOpen}, Ring, Map}, U::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}, V::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/Sheaves.jl:99
  [7] restriction_map(F::StructureSheafOfRings{CoveredScheme{QQField}, Union{AbsSpec, SpecOpen}, Ring, Map}, U::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}, V::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/Sheaves.jl:39
  [8] (::StructureSheafOfRings{CoveredScheme{QQField}, Union{AbsSpec, SpecOpen}, Ring, Map})(U::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}, V::Oscar.SimplifiedSpec{QQField, MPolyQuoLocRing{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyPowersOfElement{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/Sheaves.jl:46
  [9] restrict(f::Oscar.CoveredClosedEmbedding{CoveredScheme{QQField}, CoveredScheme{QQField}, Nothing}, DD::Covering{QQField})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/Auxiliary.jl:203
 [10] restrict(f::CoveredSchemeMorphism{CoveredScheme{QQField}, CoveredScheme{QQField}, AbsSpecMor}, inc_dom::Oscar.CoveredClosedEmbedding{CoveredScheme{QQField}, CoveredScheme{QQField}, Nothing}, inc_cod::Oscar.CoveredClosedEmbedding{CoveredScheme{QQField}, CoveredScheme{QQField}, Nothing}; check::Bool)
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/BlowupMorphism.jl:324
 [11] restrict(f::CoveredSchemeMorphism{CoveredScheme{QQField}, CoveredScheme{QQField}, AbsSpecMor}, inc_dom::Oscar.CoveredClosedEmbedding{CoveredScheme{QQField}, CoveredScheme{QQField}, Nothing}, inc_cod::Oscar.CoveredClosedEmbedding{CoveredScheme{QQField}, CoveredScheme{QQField}, Nothing})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/BlowupMorphism.jl:310
 [12] strict_transform(p::BlowupMorphism{CoveredScheme{QQField}}, inc::Oscar.CoveredClosedEmbedding{CoveredScheme{QQField}, CoveredScheme{QQField}, Nothing})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/BlowupMorphism.jl:96
 [13] relatively_minimal_model(E::Oscar.EllipticSurface{QQField, AbstractAlgebra.Generic.Frac{QQPolyRingElem}})
    @ Oscar ~/.julia/dev/Oscar/experimental/Schemes/elliptic_surface.jl:509
 [14] top-level scope
    @ REPL[41]:1

```